air/src/air/context.rs

```diff
@@ -237,6 +238,48 @@ impl<B: StarkField> AirContext<B> {
         self.num_transition_exemptions
     }
 
+    /// Returns the number of of columns needed to store the constraint composition polynomial.
+    /// This is the maximum of:
+    /// 1. The maximum evaluation degree over all transition constraints minus the degree
+    /// of the transition constraint divisor divided by trace length.
+    /// Since the degree of a constraint `C(x)` can be well approximated by
+    /// `[constraint.base + constraint.cycles.len()] * [trace_length - 1]` the degree of the
+    /// constraint composition polynomial can be approximated by:
+    /// `([constraint.base + constraint.cycles.len()] * [trace_length - 1] - [trace_length - n])`
+    /// where `constraint` is the constraint attaining the maximum and `n` is the number of
+    /// exemption points.
+    /// In the case `n = 1`, the expression simplifies to:
+    /// `[constraint.base + constraint.cycles.len() - 1] * [trace_length - 1]`
+    /// Thus, if each column is of length `trace_length`, we would need
+    /// `[constraint.base + constraint.cycles.len() - 1]` columns to store the coefficients of
+    /// the constraint composition polynomial.
+    /// This means that if the highest constraint degree is equal to `5`, the constraint
+    /// composition polynomial will require four columns and if the highest constraint degree is
+    /// equal to `7`, it will require six columns to store.
```

Comment:
minor nit: Should we move this explanation below the second point below? For example, something like this:
```
/// 1. The maximum evaluation degree over all transition constraints minus the degree
/// of the transition constraint divisor divided by trace length.
/// 2. `1`, because the constraint composition polynomial requires at least one column.
/// 
/// Since the degree of a constraint `C(x)` can be well approximated by
/// ...
```

Reply:
Done